qcodes/dataset/data_set.py

```diff
@@ -1167,6 +1176,194 @@ def __repr__(self) -> str:
+        replicating and unrolling values as needed and also handling the corner
+        case of np.array(1) kind of values
+        """
+
+        res_list: List[Dict[str, VALUE]] = []
+        all_params = inff_params.union(deps_params).union({toplevel_param})
+
+        t_map = {'numeric': float, 'text': str, 'complex': complex}
+
+        toplevel_shape = result_dict[toplevel_param].shape
+        if toplevel_shape == ():
+            # In the case of a single value, life is reasonably simple
+            res_list = [{ps.name: t_map[ps.type](result_dict[ps])
+                         for ps in all_params}]
+        else:
+            # We first massage all values into np.arrays of the same
+            # shape
+            flat_results: Dict[str, numpy.ndarray] = {}
+
+            toplevel_val = result_dict[toplevel_param]
+            flat_results[toplevel_param.name] = toplevel_val.ravel()
+            N = len(flat_results[toplevel_param.name])
+            for dep in deps_params:
+                if result_dict[dep].shape == ():
+                    flat_results[dep.name] = numpy.repeat(result_dict[dep], N)
+                else:
+                    flat_results[dep.name] = result_dict[dep].ravel()
+            for inff in inff_params:
+                if numpy.shape(result_dict[inff]) == ():
+                    flat_results[inff.name] = numpy.repeat(result_dict[dep], N)
+                else:
+                    flat_results[inff.name] = result_dict[inff].ravel()
+
+            # And then put everything into the list
+
+            res_list = [{p.name: flat_results[p.name][ind] for p in all_params}
+                        for ind in range(N)]
+
+        return res_list
+
+    @staticmethod
+    def _finalize_res_dict_standalones(
+            result_dict: Mapping[ParamSpecBase, numpy.ndarray]
+    ) -> List[Dict[str, VALUE]]:
+        """
+        Massage all standalone parameters into the correct shape
+        """
+        res_list: List[Dict[str, VALUE]] = []
+        for param, value in result_dict.items():
+            if param.type == 'text':
+                if value.shape:
+                    res_list += [{param.name: str(val)} for val in value]
+                else:
+                    res_list += [{param.name: str(value)}]
+            elif param.type == 'numeric':
+                if value.shape:
+                    res_list += [{param.name: number} for number in value]
+                else:
+                    res_list += [{param.name: float(value)}]
+            elif param.type == 'complex':
+                if value.shape:
+                    res_list += [{param.name: number} for number in value]
+                else:
+                    res_list += [{param.name: complex(value)}]
+            else:
+                res_list += [{param.name: value}]
+
+        return res_list
+
+    def flush_data_to_database(self, block: bool = False) -> None:
```

Comment:
shall we keep this private? my thinking is if it's private now, then if we decide to change/remove/move it, we don't need to do deprecation cycles and stuff.